### PR TITLE
BCDA-2596 Maintenance: Remove users table

### PR DESCRIPTION
**Pull request by dhgreene**
### Fixes [BCDA-2596](https://jira.cms.gov/browse/BCDA-2596)
Our authentication and authorization no longer has the concept of a user, but jobs and alpha tokens still reference them.  This PR removes all user references.

### Proposed Changes
- Drop `users` table
- Drop `jobs.user_id` column
- Remove all code and fixture references

### Migration Strategy
In `dev` and `test`:
- Deploy code
- `psql -f db/migrations/20200120-drop-users.sql` from deployer's laptop
- Verify success (`users` table should be gone)

In `opensbx` and `prod`:
- Deploy code
- Transfer `db/migrations/20200120-drop-users.sql` to API server in AWS environment
- SSH to server and run `psql -f 20200120-drop-users.sql`
- Verify success (`users` table should be gone)

### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications

This change should be security-neutral, removing unused database and code artifacts.

### Acceptance Validation
All tests still run ✅ 

### Feedback Requested
Are there any other references?

### CODE REVIEW

File: db/migrations/20200120-drop-users.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE jobs DROP COLUMN user_id;
+DROP TABLE users;
```
Comment (msnook) [inline db/migrations/20200120-drop-users.sql]: This migration is done outside of Gorm's auto-migrate.  Should the upcoming release ticket be updated to indicate that this script should be run in `dev`, `test`, `opensbx`, and `prod`?

**Comment by dhgreene** [comment]
- [PR created](https://github.com/CMSgov/bcda-ops/pull/398) for removing user creation Jenkins pipeline and job source
- [Onboarding doc](https://confluence.cms.gov/display/BCDA/Distributing+Beta+Partner+Credentials) updated
- [Smoke test works](https://bcda-ci.adhocteam.us/job/BCDA%20-%20Test%20-%20Integration%20Smoke/1476/) with new code (once `jobs.user_id` is allowed to be null)
- `/auth/welcome` endpoint works with newly created ACO:
Screen Shot 2020-01-20 at 3 55 11 PM

**Comment by msnook** [comment]
> * [PR created](https://github.com/CMSgov/bcda-ops/pull/398) for removing user creation Jenkins pipeline and job source
> * [Onboarding doc](https://confluence.cms.gov/display/BCDA/Distributing+Beta+Partner+Credentials) updated
> * [Smoke test works](https://bcda-ci.adhocteam.us/job/BCDA%20-%20Test%20-%20Integration%20Smoke/1476/) with new code (once `jobs.user_id` is allowed to be null)
> * `/auth/welcome` endpoint works with newly created ACO:
>   Screen Shot 2020-01-20 at 3 55 11 PM

All of these changes/tests look great!

File: db/migrations/20200120-drop-users.sql (hunk 1 above)
Comment (dhgreene) [reply db/migrations/20200120-drop-users.sql]: Yes, immediately after deployment and before running a smoke test.  Ticket is updated.

**Comment by dhgreene** [comment]
> ... we need to make sure we document the [migration] strategy for `opensbx` and `prod`.

See new `Migration Strategy` section above.

**Pull request merged**